File: December - 06/Solution.py
n = int(input("Enter n: "))
if n % 2 == 0:
    print("Magic square is only possible for odd values of n.")
else:
    magic_square = [[0] * n for _ in range(n)]

    magic_constant = n * (n * n + 1) // 2
    print(f"Magic constant: {magic_constant}")
    i = 0
    j = n // 2
    for num in range(1, n * n + 1):
        magic_square[i][j] = num
        new_i = (i - 1) % n
        new_j = (j + 1) % n
        if magic_square[new_i][new_j] != 0:
            i = (i + 1) % n
        else:
            i = new_i
            j = new_j
    for row in magic_square:
        for val in row:
            print(f"{val:3}", end=" ")
        print()

File: December - 06/solution.cpp
#include <iostream>
#include <vector>
#include <string>
#include <iomanip>

int main() {
    std::string word;
    int n = 0;
    while (std::cin >> word) {
        try {
            size_t idx;
            int val = std::stoi(word, &idx);
            n = val; 
            break; 
        } catch (...) {
            std::string num_str = "";
            for(char c : word) {
                if(isdigit(c)) num_str += c;
            }
            if(!num_str.empty()) {
                n = std::stoi(num_str);
                break;
            }
        }
    }

    if (n % 2 == 0) {
        std::cout << "Magic square is only possible for odd values of n." << std::endl;
    } else {
        long long M = (long long)n * (n * n + 1) / 2;
        std::cout << "Magic constant: " << M << std::endl;

        std::vector<std::vector<int>> magic(n, std::vector<int>(n, 0));
        int num = 1;
        int i = 0, j = n / 2;

        while (num <= n * n) {
            magic[i][j] = num;
            num++;
            int new_i = (i - 1 + n) % n;
            int new_j = (j + 1) % n;
            if (magic[new_i][new_j]) {
                i = (i + 1) % n;
            } else {
                i = new_i;
                j = new_j;
            }
        }

        for (int r = 0; r < n; ++r) {
            for (int c = 0; c < n; ++c) {
                std::cout << magic[r][c] << (c == n - 1 ? "" : "   ");
            }
            std::cout << std::endl;
        }
    }
    return 0;
}
